Java/BST.java

```diff
@@ -0,0 +1,349 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/** A class to denote a node of the BST.
+ */
+class Node {
+    /** AVL Node hold's this value.
+     */
+    public int value;
+
+    /** Pointer to left child of the BST Node.
+     */
+    public Node left;
+
+    /** Pointer to right child of the BST Node.
+     */
+    public Node right;
+
+    /** Height of the BST Node.
+     */
+    public int height;
+
+    /** Creates a single BST Node with the given value.
+     * @param value The value of the node.
+     */
+    Node(int value) {
+        this.value = value;
+        this.left = null;
+        this.right = null;
+        this.height = 1;
+    }
+
+    @Override
+    public String toString() {
+        return "Node{" +
+                "value=" + value +
+                ", height=" + height +
+                '}';
+    }
+}
+
+
+public class BST {
+    public Node root;
+
+    public BST() {
+        this.root = null;
+    }
+
+    public int height() {
+        if (this.root != null)
+            return getHeight(this.root);
+        return 0;
+    }
+
+    private int getHeight(Node node) {
+        if (node != null)
+            return node.height;
+        return 0;
+    }
+
+    /**
+     * Inserts the given value into the AVL Tree and updates the root node.
+     *
+     * @param value Value to be inserted.
+     */
+    public void insert(int value) {
+        this.root = insert(this.root, value);
+    }
+
+    private Node insert(Node node, int value) {
+        if(node == null)
+            return new Node(value);
+        else if(value < node.value)
+            node.left = insert(node.left, value);
+        else
+            node.right = insert(node.right, value);
+
```

Comment:
You're allowing insertion of duplicates in right subtree. BST don't allow duplicates.

Reply:
Okay, I have removed duplicates from the Implementation.